src/HTSPVFS.cpp

```diff
@@ -130,70 +88,40 @@ void CHTSPVFS::Close ( void )
   m_offset = 0;
   m_fileId = 0;
   m_path   = "";
-
-  Reset();
-}
-
-void CHTSPVFS::Reset()
-{
-  CLockObject lock(m_mutex);
-  m_buffer.reset();
-  m_bHasData = false;
-  m_bSeekDone = true;
-  m_currentReadLength = INITAL_READ_LENGTH;
-  m_seekCondition.Signal();
 }
 
 int CHTSPVFS::Read ( unsigned char *buf, unsigned int len )
 {
-  ssize_t ret;
-  CLockObject lock(m_mutex);
-
   /* Not opened */
   if (!m_fileId)
     return -1;
 
-  m_seekCondition.Wait(m_mutex, m_bSeekDone, 5000);
-
-  /* Signal that we need more data in the buffer. Reset the read length to the 
-     requested length so we don't wait unnecessarily long */
-  if (m_buffer.avail() < len)
-  {
-    m_bHasData = false;
-    m_currentReadLength = len;
-    m_condition.Broadcast();
-  }
-
-  /* Wait for data */
-  m_condition.Wait(m_mutex, m_bHasData, 5000);
-
   /* Read */
-  ret = m_buffer.read(buf, len);
-  m_offset += ret;
-  return (int)ret;
+  int read = SendFileRead(buf, len);
```

Comment:
Should be checking for return value here. I assume we're relying on caller to do the right thing.

But at the very least we shouldn't be doing m_offset += -1; in error case.